### PR TITLE
 Fix handling of NTAG OTP and LOCK bytes

### PR DESCRIPTION
**Pull request by AdamLaurie**

### CODE REVIEW

File: utils/nfc-mfultralight.c [hunk 1]
```diff
@@ -376,13 +388,19 @@ write_card(bool write_otp, bool write_lock, bool write_uid)
...
+         (iNTAGType == NTAG_215 && page == 0x82) || \
+         (iNTAGType == NTAG_216 && page == 0xe2)) && (!write_lock)) {
       printf("s");
       uiSkippedPages++;
       continue;
     }
-    if ((page == 0x3) && (!write_otp)) {
+    // NTAG doesn't have OTP blocks but Static Lock Bytes are OTP
+    if (((iNTAGType == NTAG_NONE && page == 0x3) || \
+        (iNTAGType && page == 0x2)) && (!write_otp)) {
```
Comment (doegox) [inline utils/nfc-mfultralight.c]: in (iNTAGType && page == 0x2)
please write iNTAGType != NTAG_NONE to not implicitly depend on NTAG_NONE value.

Comment (AdamLaurie) [reply utils/nfc-mfultralight.c]: OK. However, I've just realised that the behaviour of the Ultralight LOCK bytes is the same - they are also OTP (if you set them), so maybe my logic/phrasing is now rather confusing. The EV1 Ultralight also has a trailing Dynamic lock byte section... I think I have to do this differently! :) Stand by!!!

File: utils/nfc-mfultralight.c [hunk 2]
```diff
@@ -388,19 +393,23 @@ write_card(bool write_otp, bool write_lock, bool write_uid)
...
-    if (((iNTAGType == NTAG_NONE && page == 0x2) || \
-         (iNTAGType == NTAG_213 && page == 0x28) || \
-         (iNTAGType == NTAG_215 && page == 0x82) || \
-         (iNTAGType == NTAG_216 && page == 0xe2)) && (!write_lock)) {
+    if ((!write_lock) && page == 0x2) {
+      printf("s");
+      uiSkippedPages++;
+      continue;
+    }
+    // NTAG doesn't have OTP blocks
```
Comment (doegox) [inline utils/nfc-mfultralight.c]: Scoop: yes NTAG also have OTP :) Capability container at page 0x02 is just another name but still with OTP.

Comment (AdamLaurie) [reply utils/nfc-mfultralight.c]: Ah, I knew I should have checked! :P  OK, I'll tweak it (again!). Lol.

**Comment by doegox** [comment]
Seems good to me, thanks!

**Pull request merged**